### DISCUSSION TITLE
Temporary .csproj created

### DISCUSSION THREAD

**Discussion by iacore** [fable-compiler/Fable#3352] [Q&A]
When running `fable`, a temporary `projectname.csproj` is created and then deleted.

Then the command `dotnet restore projectname.csproj` is run.

Is it necessary? `dotnet restore projectname.fsproj` works fine.

**Comment by MangelMaxime** [comment]
Hello,
The temporary csprof is created in order to resolve the dependencies list. Msbuild has a better support for csproj file and doing it that way allow Fable 4 to support .Net 7. 

I also think that the hope is that future version of .Net will be easier to support as for example it is still not supported yet but the custom project cracker in Fable 3.

**Comment by MangelMaxime** [comment]
Here is the link to the issue where the csproj usage has been introduced 

https://github.com/fable-compiler/Fable/issues/3260
